### DISCUSSION TITLE
Custom language notebook syntax highlight with LSP Language Server Protocol

### DISCUSSION THREAD

**Discussion by pl7y** [microsoft/vscode-discussions#588] [Extension Development QnA] [answered]
I am making a custom language extension for VSCode (basically semantic highlight of the language using a custom LSP), and I want to support such language in VSCode notebooks in the same extension. Semantic highlight using LSP works perfectly when editing a file, and notebooks execute correctly the language.
The problem is that **semantic highlight is not applied when editing notebook cells**, it seems that the notebook doesn't even try to do it as if it was not configured to do so. I couldn't find any hint in the docs.

**Comment by r3m0t** [comment, marked as answer]
Did you use the notebookType field on DocumentFilter? Can you share the DocumentSelector used in the LSP client options.

**Reply by pl7y** [reply]
Excellent suggestion, I was filtering for the `file` type:

```typescript
	// Options to control the language client
	const clientOptions: LanguageClientOptions = {
		// Register the server for plain text documents
		documentSelector: [{ scheme: 'file', language: 'langname' }],
		synchronize: {
			// Notify the server about file changes to '.clientrc files contained in the workspace
			fileEvents: workspace.createFileSystemWatcher('**/.clientrc')
		}
	};


```
